### DISCUSSION TITLE
When using the gRPC server in standalone way, the `servletContext` is null.

### DISCUSSION THREAD

**Discussion by liweinan** [resteasy/resteasy-grpc#16] [General]
If I use the `main` method of `Greet_Server` to start the gRPC server in standalone way:

```java
    /**
    * Main launches the server from the command line.
    */
    public static void main(String[] args) throws Exception, InterruptedException {
        final Greet_Server server = new Greet_Server();
        server.start();
        server.blockUntilShutdown();
    }
```

And do a request to the server, then the `servletContext` is null:

![26851690910932_ pic](https://github.com/resteasy/resteasy-grpc/assets/201907/42a2f8af-ebda-46dc-8298-5052ced56ef9)

Is there any suggestion to use the bridge project in a non-integrated environment (without WildFly)?

**Comment by liweinan** [comment]
@ronsigal Do you have any suggestions on this?

**Comment by ronsigal** [comment]
Heyyy @liweinan, actually, I've rewritten this part a bit. Now a ServletContext is grabbed in GrpcHttpServletDispatcher.init(), which, I believe, is called with the WAR is deployed. On the other hand, Greet_Server.startContext() is used to grab an HttpServletRequest, which can be used as a delegate for methods like HttpServletRequest.login().  This is still a work in progress: I think I've got the code right, more or less, but I've had some Arquillian trouble with writing tests. I just got the Arquillian part working today, so I hope to have a pull request ready soon. Sorry for the confusion.

**Comment by liweinan** [comment]
@ronsigal Cool! Looking forward to this :D

**Comment by liweinan** [comment]
@ronsigal With my understanding, when this gRPC server is started in a standalone way, the gRPC service itself is deployed in `grpc-server`, which is Netty based, and doesn't have a Servlet container related with it, so the WAR file has to be deployed in a Servlet container like WildFly or Tomcat.

But if I do a request to the gRPC server, the servlet context in the backend must be fetched from the Servlet container. 


In WildFly, this has no problem, because it's an integrated environment: The gRPC subsystem provides the gRPC service, and the servlet container is in the same context, so the `servletContext` will not be null. But how will this be implemented in standalone environment?

The problem is that in standalone environment, the gRPC server and the Servlet server are two independent containers, so how can the servlet and context(other components) be passed across the two independent containers? Currently the gRPC service and the backend Servlet based services are integreated together, which means the gRPC service has to fetch the `servletContext` to work. I guess in the future the "front" gRPC service and the "back" Servlet service has to be decoupled, am I right?

**Comment by liweinan** [comment]
@ronsigal I haven't figured out how to make the whole thing work without a WlildFly environment, I'm not sure whether it's my usage problem. If you know something about this topic, could you please teach me on this? Thanks! :D
